### PR TITLE
Add chat.quadrix.Quadrix

### PR DESCRIPTION
**Pull request by alariej**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. (I do need `--device=all`, please see below)
- [x] All assets referenced in the manifest are redistributable by any party.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### Additional info:

- I am the developer and copyright holder of quadrix
- quadrix is an open source project hosted on github at https://github.com/alariej/quadrix
- I own https://quadrix.chat
- A `--device=all` permission is required to access the camera on the host machine (for the videoconferencing function)
- quadrix is already available as a snap (https://snapcraft.io/quadrix), and also on all main mobile and desktop platforms
- all of quadrix desktop apps are Electron apps
- I have successfully built the quadrix flatpak on my dev machine (Ubuntu 21.10)
- I am an absolutely newbie to flatpak and flathub

### CODE REVIEW

File: chat.quadrix.Quadrix.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-download
+build-options:
+  append-path: /usr/lib/sdk/node16/bin
+  cflags: -O2 -g
+  cxxflags: -O2 -g
```
Comment (hfiguiere) [inline chat.quadrix.Quadrix.yml:24]:
```suggestion
```

File: chat.quadrix.Quadrix.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+app-id: chat.quadrix.Quadrix
+branch: stable
```
Comment (hfiguiere) [inline chat.quadrix.Quadrix.yml:2]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build chat.quadrix.Quadrix

File: chat.quadrix.Quadrix.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      env:
+        XDG_CACHE_HOME: /run/build/quadrix/flatpak-node/cache
+        npm_config_cache: /run/build/quadrix/flatpak-node/npm-cache
+        npm_config_nodedir: /usr/lib/sdk/node16
+        npm_config_offline: 'true'
+    subdir: main
+    sources:
+      - type: dir
+        path: ..
+        dest: main
```
Comment (hfiguiere) [inline chat.quadrix.Quadrix.yml:37]: so where does it get the sources?

I missed that on first review, but that's the reason the build failed.

Comment (alariej) [reply chat.quadrix.Quadrix.yml:37]: Hmmm, the source is on my local machine. I was wondering what was being built by the flathubbot. Obviously the manifest as it is cannot be enough to build the app.

Does it mean I cannot upload a local build to flathub directly, and flathub has to build the whole app by itself? Is it something specific to Electron apps? Would it be easier to use a .deb file as the source?

File: chat.quadrix.Quadrix.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - generated-sources.json
+      - type: file
+        path: chat.quadrix.Quadrix.metainfo.xml
```
Comment (hfiguiere) [inline chat.quadrix.Quadrix.yml:40]: this file is missing

File: chat.quadrix.Quadrix.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      - type: file
+        path: chat.quadrix.Quadrix.desktop
+      - type: file
+        path: chat.quadrix.Quadrix.svg
```
Comment (hfiguiere) [inline chat.quadrix.Quadrix.yml:44]: this file is missing

**Review by hfiguiere — CHANGES_REQUESTED** [review]
more things missing

File: chat.quadrix.Quadrix.yml (hunk 3 above)
Comment (hfiguiere) [reply chat.quadrix.Quadrix.yml:37]: The manifest is to build the flatpak and this is done on flathub infrastructure.

You can put a source as git, a tarball, etc. as per the documentation.

**Comment by alariej** [comment]
OK thanks, I will upload the missing files in the next hours.

Quick question: Is it not possible to submit a complete flatpak build folder (which I already have on my local dev machine) instead of having flathub re-build the whole app?

**Comment by alariej** [comment]
OK thanks. What about using a pre-compiled .deb file as source? Is it possible? Is the submission process then simpler, not having to re-build the Electron app within flathub?

**Comment by alariej** [comment]
I'm now trying to build the flatpak locally with a source of type "archive":

 ```
   sources:
      - type: archive
        url: https://github.com/alariej/quadrix/archive/master.tar.gz
        sha256: a4d17c66f10e32675ad1a7f8607cc28cae70fb5877baa0fc32633ce1b7d0812c
        dest: main
        ...
```
The problem is that my project uses git sources as dependencies, and flatpak-builder fails when trying to retrieve them:

```
npm ERR! code ENOTCACHED
npm ERR! request to https://codeload.github.com/alariej/ReSub/tar.gz/693bb7cadf0562375654453e52fe3c2417e746b8 failed: cache mode is 'only-if-cached' but no cached response is available.
```
I'm kind of stuck here at the moment. Have you seen this problem before?

**Comment by hfiguiere** [comment]
Not sure what the proble is here, but if you git repo use submodules then you need to either get the by hand or use a "source": "git" for the main source that will recursively fetch the submodules as needed.

Also `https://github.com/alariej/quadrix/archive/master.tar.gz` as source is always a bad idea because you should use a specific commit.

**Comment by alariej** [comment]
Thanks for the hint on the specific commit, I'll change it when / if I finally succeed at producing a flatpak from a github archive.

I thought about using `source: git`, but as I am reading the docs on `flatpak-builder` and `flatpak-node-generator`, the git dependencies should be downloaded automatically by `flatpak-builder` to a specific folder (/flatpak-node/git-packages/) in the build folder (I can see that, so far so good), and the package*.json files should then be patched (patch.sh script seems to contain the correct info) so that the dependencies point to that folder. But that last step doesn't seem to work (the package*.json files don't get patched) and that's why `flatpak-builder` is trying to download the git dependencies again during the `npm install --offline` sequence.

**Comment by alariej** [comment]
bot, build chat.quadrix.Quadrix

**Comment by alariej** [comment]
bot, build chat.quadrix.Quadrix

File: chat.quadrix.Quadrix.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+modules:
+  - name: quadrix
+    buildsystem: simple
+    skip-arches:
+      - aarch64
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://quadrix.chat/releases/deb/amd64/quadrix_0.6.5_amd64.deb
```
Comment (hfiguiere) [inline chat.quadrix.Quadrix.yml:28]: so what about aarch64 users?

Comment (alariej) [reply chat.quadrix.Quadrix.yml:28]: OK will remove `skip-arches` and build a _aarch64_ .deb this weekend :-)

Comment (alariej) [reply chat.quadrix.Quadrix.yml:28]: But how can I skip the _aarch64_ test build until I have a _aarch64_ .deb file? The `skip-arches` option didn't seem to do anything.

Comment (hfiguiere) [reply chat.quadrix.Quadrix.yml:28]: https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

**Comment by alariej** [comment]
bot, build chat.quadrix.Quadrix

**Comment by alariej** [comment]
@hfiguiere Not sure about the next steps. Is the app now ready to be published on flathub.org? Anything to do on my side? Thanks.

**Pull request closed without merging**